sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/models/AttestationPolicySetOptions.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.security.attestation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.Context;
+
+/**
+ * <p>
+ * AttestationPolicySetOptions represent the parameters sent to the
+ * {@link com.azure.security.attestation.AttestationAdministrationClient#getAttestationPolicy(AttestationType)}
+ * or {@link com.azure.security.attestation.AttestationAdministrationClient#getAttestationPolicyWithResponse(AttestationType, Context)} API.
+ * </p><p>
+ * Each {@link AttestationPolicySetOptions} object expresses the options to verify the response
+ * from the attestation service.
+ * </p>
+ */
+@Fluent
+public final class AttestationPolicySetOptions {
+
+    private AttestationTokenValidationOptions validationOptions;
+    private String policy;
+    private AttestationSigningKey signer;
+
+    /**
+     * Sets the options used to validate attestation tokens returned from the service.
+     * @param validationOptions Token Validation options to be used to enhance the validations
+     *                          already performed by the SDK.
+     * @return this {@link AttestationPolicySetOptions} object.
+     */
+    public AttestationPolicySetOptions setValidationOptions(AttestationTokenValidationOptions validationOptions) {
+        this.validationOptions = validationOptions;
+        return this;
+    }
+
+    /**
+     * Returns the options used for token validation.
+     * @return attestation token validation options.
+     */
+    public AttestationTokenValidationOptions getValidationOptions() {
+        return validationOptions;
+    }
+
+    /**
+     * Gets the attestation policy which will be used to generate an policy set request.
+     * @return Attestation Policy Token associated with this request.
+     */
+    public String getAttestationPolicy() {
```

Comment:
>getAttestationPolicy [](http://example.com/codeflow?start=18&length=20)

Should this be named setAttestationPolicy analogous to getAttestationPolicy?

Reply:
Yes. I was thinking about removing the word "Attestation" everywhere since it's embedded in the class name (`AttestationPolicySetOptions` - Options for "attestation policy set").

Thoughts?

Reply:
We generally choose to be verbose on Java. But if we have already set a precedence in .Net with the property name we will have to follow that here too. We try to be consistent in naming in .Net and Java wherever possible.

Reply:
.Net doesn't have a corresponding construct because it doesn't do the whole "fluent options" thing. For .Net, the properties in the PolicySetOptions are all parameters to the method. But the Java pattern appears to be to provide an options object instead of parameters.